numpy/f2py/__init__.py

```diff
@@ -28,12 +28,16 @@ def compile(source,
             extension='.f'
            ):
     """
-    Build extension module from processing source with f2py.
+    Build extension module from a Fortran 77 source string with f2py.
 
     Parameters
     ----------
-    source : str
+    source : str or bytes
         Fortran source of module / subroutine to compile
+
+        .. versionchanged:: 1.16.0
```

Comment:
1.16.1

Reply:
Was this supposed to work in 1.16.0 and just a bug?

Reply:
I think it was an unintentional change. Before 1.16.0 it had to be bytes, in 1.16.0 it was changed to str by a bug. After this PR it will accept both